src/software/thunderscope/binary_context_managers/game_controller.py

```diff
@@ -102,7 +107,23 @@ def refresh(self):
             )
             manual_command = self.command_override_buffer.get(return_cached=False)
 
-    def next_free_port(self, port: int = 40000, max_port: int = 65535) -> None:
+    def is_valid_port(self, port):
+        """
+        Check if a port is available 
+
+        :param port: the port we are checking
+        :return: True if the port is available, False otherwise
+        """
+        sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+
+        try:
+            sock.bind(("", port))
+            sock.close()
+            return True
+        except OSError:
+            return False
+
+    def next_free_port(self, start_port: int = 40000, max_port: int = 65535) -> None:
```

Comment:
This seems to have been a mistake from before:
```suggestion
    def next_free_port(self, start_port: int = 40000, max_port: int = 65535) -> int:
```

Reply:
The function doc also needs to be updated to have parameter `start_port` as opposed to just `port`